pkg/controller/machineset/delete_policy.go

```diff
@@ -87,6 +88,9 @@ func randomDeletePolicy(machine *v1beta1.Machine) deletePriority {
 	if machine.Status.ErrorReason != nil || machine.Status.ErrorMessage != nil {
 		return betterDelete
 	}
+	if machine.Status.Phase == nil || *machine.Status.Phase != "Running" {
+		return preferDelete
+	}
```

Comment:
Might it be better to rely on the fields that cause the phases to change, for example, if you checked that there is no noderef set here, that would cover both these cases, a node ref being set causes the phase to become Running.

We are trying to avoid logic based on phases where possible

Reply:
I can certainly do that. I didn't do that in the first instance because it felt like duplicating the logic that sets the phase, and a potential future maintenance issue. A Condition would be nice here, tbh.

Are you sure you want this?

Reply:
Yes please. It is in keeping with the rest of the logic here which represents other phases (eg the error one being the failed phase).

Reply:
Yeah, that logic seems trivial enough. Updated.